tests/test_integration.py

```diff
@@ -1804,3 +1804,69 @@ async def test_suite_not_a_suite_directory(envsonschema, tmp_path):
         exit_code=-1,
     )
     assert re.search(r"does not contain .* cases", stderr)
+
+
+@pytest.mark.asyncio
+async def test_dialect_warning_validate(envsonschema, tmp_path):
+    tmp_path.joinpath("schema.json").write_text(
+        '{"$schema": "https://json-schema.org/draft/2020-12/schema"}',
+    )
+    tmp_path.joinpath("a.json").write_text("12")
+
+    _, stderr = await bowtie(
+        "validate",
+        "-D",
+        "7",
+        "-i",
+        envsonschema,
+        tmp_path / "schema.json",
+        tmp_path / "a.json",
+    )
+
+    assert re.search(
+        str.join(
+            r"",
+            [
+                r"\[Warning\s*\]\s*The \$Schema Poperty Refers To ",
+                r"'Draft \d{4}-\d{2}' ",
+                r"While The Dialect Argument Is 'Draft \d+'\n",
+            ],
+        ),
+        stderr.title(),
```

Comment:
You should see big alarm bells when you write what you wrote here, it's a very strange message to be checking against, which comes from you calling `.title()`, for reasons I'm not sure.

Note that your message looks really strange, and references `$Schema` rather than `$schema`, which would be very wrong if we showed, as that keyword doesn't exist.

Reply:
Sir the thing is the stderr is making the error message capitalised
even though in the warning that I display in the terminal it's shown as `$schema`.
I can document it in the code if necessary.

Reply:
Then call `lower` as all the other tests do, not `title`.